File: src/main/java/com/market/ddanggyeo/security/UserDetailsServiceImpl.java
package com.market.ddanggyeo.security;

import com.market.ddanggyeo.account.entity.Account;
import com.market.ddanggyeo.account.repository.AccountRepository;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class UserDetailsServiceImpl implements UserDetailsService {

    private AccountRepository accountRepository;

    UserDetailsServiceImpl(AccountRepository accountRepository) {
        this.accountRepository = accountRepository;
    }

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        Account account = accountRepository.findByEmail(username);
        if (account == null) {
            throw new UsernameNotFoundException("존재하지 않은 계정입니다.");
        }

        return new UserDetailsImpl(account.getEmail(), account.getPassword());
    }
}

File: src/main/kotlin/com/market/ddanggyeo/security/UserDetailsServiceImpl.kt
package com.market.ddanggyeo.security

import com.market.ddanggyeo.account.entity.Account
import com.market.ddanggyeo.account.repository.AccountRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserDetailsServiceImpl(private val accountRepository: AccountRepository): UserDetailsService {
    override fun loadUserByUsername(username: String): UserDetails {
        val account: Account = accountRepository.findByEmail(username)?:
            throw UsernameNotFoundException("존재하지 않은 계정입니다.")
        return UserDetailsImpl(account)
    }
}